src/Our.Umbraco.InnerContent/PropertyEditors/SimpleInnerContentPreValueEditor.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Collections.Generic;
+using Newtonsoft.Json.Linq;
+using Our.Umbraco.InnerContent.Helpers;
+using Umbraco.Core;
+using Umbraco.Core.Models;
+using Umbraco.Core.PropertyEditors;
+
+namespace Our.Umbraco.InnerContent.PropertyEditors
+{
+    public class SimpleInnerContentPreValueEditor : InnerContentPreValueEditor
+    {
+        [PreValueField("contentTypes", "Content Types", "~/App_Plugins/InnerContent/views/innercontent.doctypepicker.html", Description = "Select the content types to use as the data blueprint.")]
+        public string[] ContentTypes { get; set; }
+
+        [PreValueField("maxItems", "Max Items", "number", Description = "Set the maximum number of items allowed in this stack.")]
+        public string MaxItems { get; set; }
+
+        [PreValueField("singleItemMode", "Single Item Mode", "boolean", Description = "Set whether to work in single item mode (only the first defined Content Type will be used).")]
+        public string SingleItemMode { get; set; }
+
+        [PreValueField("hideLabel", "Hide Label", "boolean", Description = "Set whether to hide the editor label and have the list take up the full width of the editor window.")]
+        public string HideLabel { get; set; }
+
+        [PreValueField("disablePreview", "Disable Preview", "boolean", Description = "Set whether to disable the preview of the items in the stack.")]
+        public string DisablePreview { get; set; }
+
+        public override IDictionary<string, object> ConvertDbToEditor(IDictionary<string, object> defaultPreVals, PreValueCollection persistedPreVals)
+        {
+            if (persistedPreVals.IsDictionaryBased)
+            {
+                var dict = persistedPreVals.PreValuesAsDictionary;
+                if (dict.TryGetValue("contentTypes", out PreValue contentTypes) && string.IsNullOrWhiteSpace(contentTypes.Value) == false)
+                {
+                    var items = JArray.Parse(contentTypes.Value);
```

Comment:
I wonder if, like the property editor code, rather than this explicitly calling TryEnsureContentTypeGuids here, should lines 34 to 38 be part of the base class? and we have a ConvertInnerContentDbToEditor that handles any "content types conversion"?

All other properties I think are implementation specific, but the content types are explicitly IC related

Reply:
Cool, I'll update the PRs 👍

Reply:
I tried to add a `ConvertInnerContentDbToEditor`, but got a bit confused how it should work.
We could merge the PR in (pending the other change) and refactor it from there? or did you want to get it sorted in this PR?

Reply:
If we get the non-contenttype properties moved back to stacked content so SimpleInnerContentPreValueEditor is just handling contenttypes, then I can guess we can continue to play with how that is internally structured without affecting anything, so happy for it to get merged in.

Reply:
Cool - working on it right this moment.